application/src/main/java/org/togetherjava/tjbot/features/help/PinAnswerCommand.java

```diff
@@ -0,0 +1,73 @@
+package org.togetherjava.tjbot.features.help;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+
+import org.togetherjava.tjbot.features.BotCommandAdapter;
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.MessageContextCommand;
+
+public final class PinAnswerCommand extends BotCommandAdapter implements MessageContextCommand {
+    private static final String COMMAND_NAME = "pin-answer";
+    private static final int MAX_PINNED_ANSWERS = 10;
+    private int count = 0;
+
+    public PinAnswerCommand() {
+        super(Commands.message(COMMAND_NAME), CommandVisibility.GUILD);
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        Message originalMessage = event.getTarget();
+        User commandInvoker = event.getUser();
+
+        if (!(originalMessage.getChannel() instanceof ThreadChannel threadChannel)) {
+            replyNotInThread(event);
+            return;
+        }
+
+        if (!isThreadCreator(commandInvoker, threadChannel)) {
+            replyNotThreadCreator(event);
+            return;
+        }
+
+        if (count >= MAX_PINNED_ANSWERS) {
+            replyMaxPinsReached(event);
+            return;
+        }
+
+        pinMessage(event, originalMessage);
+    }
+
+    private boolean isThreadCreator(User user, ThreadChannel thread) {
+        return user.getIdLong() == thread.getOwnerIdLong();
+    }
```

Comment:
just a nitpick, `threadOwner` sounds a bit better to me. But it's no worries if you wanna keep it as it is.